### ISSUE TITLE
Upgrade to Spring Batch 5

### ISSUE THREAD

**Issue by mdeinum** [spring-projects/spring-batch-extensions#111]
Spring Batch 5 has been released some time ago, it is time to release a version that is compatible with Spring Batch 5.

**Comment by krish300** [comment]
@mdeinum I am trying to upgrade one of my spring-boot projects to spring-boot-3.1.3. I have spring-batch-excel-0.1.1 as dependency. May I know if you have any plans to do this upgrade in near future

**Comment by mdeinum** [comment]
Thanks for your comments and usage of this library. I'm currently a bit swamped in other work. So a PR is welcome if you have the time. As there is no real changes in the `javax` -> `jakarta` namespace for this I expect it to be a fairly straightforward upgrade (maybe some method signatures need to change).

**Comment by mdeinum** [comment]
I did some quick tests and the project seems to work without changes with both Java17 and Spring Batch 5. So there should nothing be preventing you from upgrading (which is probably also the reason I didn't create a dedicated version for Spring Batch 5, as it just works (tm) ).

I do have a branch in my own repository (https://github.com/mdeinum/spring-batch-extensions/tree/spring-batch-excel-batch5) which you can checkout. The only things changed there are the dependencies and some build related artifacts. The code has remained unchanged. 

So in short it should work fine with Spring Batch 5, JDK17 and Apache POI 5.2.3 (the latest version). At least according to the testsuite we have in place.

**Comment by krish300** [comment]
You are right. Works well with "Spring Batch 5, JDK17 and Apache POI 5.2.3". Thanks for the prompt response and all the amazing work you do. 

btw, do you think I can still PR or you will PR the changes you did in your repo. I never contributed before. :)

**Comment by mdeinum** [comment]
Well as it works we don't really need a PR. There are some things in that PR that I will move to main (upgrades to plugins and how things are managed). 

For the compatibility I'll update the documentation to state that it works with Spring Batch 4.3, 5.0 (and I suspect also with  5.1 which is upcoming).
